### ISSUE TITLE
Add ability to show and hide grid lines in a spreadsheet

### ISSUE THREAD

**Issue by zahrevsky** [burnash/gspread#1197]
**Is your feature request related to a problem? Please describe.**
Add ability to show and hide grid lines in a spreadsheet. This can be done via batch_update using the following request body:
```
     [{
         'updateSheetProperties': {
                "fields": "gridProperties.hideGridlines",
                "properties": {
                    "gridProperties": {
                        "hideGridlines": True,
                    },
                    "sheetId": self.sheet.id,
                },
            },
        }]
```
Docs for gridProperties: https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/sheets#GridProperties

**Additional context**
There's a similar feature request about hiding rows/columns (https://github.com/burnash/gspread/issues/1006), which is solved in a pull request https://github.com/burnash/gspread/pull/1012/. With hiding gridlines there are only two differences:
1. batch_update body is `updateSheetProperties`, not `updateDimensionProperties`
2. `hide` and `unhide` should be methods of a `Spreadsheet` class, not a `Worksheet`

**Comment by lavigne958** [comment]
Hi thank you for proposing this feature, this can be done i presume I still need to check and test .

**Comment by alifeee** [comment]
> hide and unhide should be methods of a Spreadsheet class, not a Worksheet

Why should this be? On Google Sheets on the web, gridlines are shown/hidden per-worksheet.

**Comment by zahrevsky** [comment]
> > 2. hide and unhide should be methods of a Spreadsheet class, not a Worksheet
> 
> Why should this be? On Google Sheets on the web, gridlines are shown/hidden per-worksheet.

Oops, I've got it wrong. Yes, this should be a Worksheet's method, not a Spreadsheet's

**Comment by alifeee** [comment]
I have implemented a version of this #1202
